addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -201,12 +201,37 @@ function ActivityFactory({ Entity }) {
             });
         }
 
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsCurrentPartnerAssignee() {
+            if (!this.assigneePartner || !this.messagingCurrentPartner) {
+                return false;
+            }
+            return this.assigneePartner === this.messagingCurrentPartner;
```

Comment:
I would use `this.assignee.partner` and `this.env.messaging.currentPartner` instead.
Reason is that `assigneePartner` and `messagingCurrentPartner` are related fields just because we need them in order to express compute dependencies, but that's a small annoyance for the moment because other than that they are not necessary. Having them in field definition is already noisy enough, so I would just avoid using them explicitly.

Reply:
@seb-odoo as you suggest the inverse, do you agree ?

@alexkuhn then I have to switch back `!this.assigneePartner` to `!this.assignee || !this.assignee.partner` or can I assume `this.assignee` is defined ?

Reply:
Don't assume `this.assignee` is defined, so former proposition.

Reply:
Well this is partially why I suggested it, using related is automatically taking care of "undefined in the middle".

If we are sure to remove the need of intermediary related field at some point (so allow compute dependencies to be relations), then I'm ok to revert to the other syntax.

But then we would also need the concept of (empty) recordsets if we want "undefined in the middle" to also be taken care of automatically. Until then it has to be explicit.

Both of those features are just nice to have though, there are a lot more important things first.

Reply:
Ideally ideally, compute dependencies should not even have to be defined. Similar to useStore should not be defined at all.

At each render/compute, starting from the first, the code should be able to tell which fields/relations were used during this time, and flag them as dependencies automatically.

Reply:
Each has pros/cons, so all things considered I don't really mind using them now.

Just feel that `assigneePartner` and `messagingCurrentPartner` are basically aliases of `assignee.partner` and `messaging.currentPartner`, and that's why I would prefer using later over former.

I also think falsy `assigneePartner` is ambiguous, because it could either mean no assignee or no partner related to assignee, and the nuance is necessary sometimes. Relying on `undefined` in any of these cases is similar to using optional chaining (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), which is a debatable feature (kinda explains why it took so long for Chrome and Firefox to just recently support this easy-to-implement feature, because there were a lot of criticisms to encourage bad coding practices).

Reply:
> Just feel that assigneePartner and messagingCurrentPartner are basically aliases

Well yes, this is the only use for related fields, they could be deleted, replaced by the relation used explicitly in the code, and everything else would work the same.

> I also think falsy assigneePartner is ambiguous, because it could either mean no assignee or no partner related to assignee, and the nuance is necessary sometimes. 

This is true however, and I have real cases on my notif branch were the related field cannot be used at all in the compute due to a similar ambiguity.